### PR TITLE
better validation of auth api inputs

### PR DESCRIPTION
**Pull request by invisig0th**

### CODE REVIEW

File: synapse/lib/hiveauth.py [hunk 1]
```diff
@@ -574,6 +590,7 @@ def getRules(self, gateiden=None):
         return list(gateinfo.get('rules', ()))
 
     async def setRules(self, rules, gateiden=None, nexs=True):
+        reqValidRules(rules)
         return await self._setRulrInfo('rules', rules, gateiden=gateiden, nexs=nexs)
 
     async def addRule(self, rule, indx=None, gateiden=None, nexs=True):
```
Comment (vEpiphyte) [inline synapse/lib/hiveauth.py:596]: You need to protect do validation around addrule.

`reqValidRules((rule,))` would do it.

File: synapse/lib/hiveauth.py [hunk 2]
```diff
@@ -6,9 +6,21 @@
...
 logger = logging.getLogger(__name__)
 
+reqValidRules = s_config.getJsValidator({
+    'type': 'array',
+    'items': {
+        'type': 'array',
+        'items': [
+            {'type': 'boolean'},
+            {'type': 'array', 'items': {'type': 'string'}},
+        ],
```
Comment (vEpiphyte) [inline synapse/lib/hiveauth.py:20]:
```
  "minItems": 2,
  "maxItems": 2
```
^ Should also validate the length of each item in our array

File: synapse/tests/test_lib_hiveauth.py [hunk 3]
```diff
@@ -340,3 +340,25 @@ async def test_hive_auth_persistence(self):
...
+
+    async def test_hive_auth_invalid(self):
+
+        async with self.getTestCore() as core:
+            with self.raises(s_exc.BadArg):
+                await core.auth.rootuser.setName(1)
+            with self.raises(s_exc.BadArg):
+                await core.auth.allrole.setName(1)
+            with self.raises(s_exc.SchemaViolation):
+                await core.auth.rootuser.addRule('vi')
```
Comment (vEpiphyte) [inline synapse/tests/test_lib_hiveauth.py:352]: bad test since its not using the schema

File: synapse/lib/hiveauth.py (hunk 2 above)
Comment (invisig0th) [reply synapse/lib/hiveauth.py:20]: the positional specification does this

**Pull request merged**